keras/layers/convolutional.py

```diff
@@ -11,7 +11,90 @@
+
+        self.activation = activations.get(activation)
+        assert border_mode in {'valid', 'same'}, 'border_mode must be in {valid, same}'
+        self.border_mode = border_mode
+
+        self.W_regularizer = regularizers.get(W_regularizer)
+        self.b_regularizer = regularizers.get(b_regularizer)
+        self.activity_regularizer = regularizers.get(activity_regularizer)
+
+        self.W_constraint = constraints.get(W_constraint)
+        self.b_constraint = constraints.get(b_constraint)
+
+        self.bias = bias
+        self.input_spec = [InputSpec(ndim=2 + len(dimensions))]
+        assert dim_ordering in {'tf', 'th'}, 'dim_ordering must be in {tf, th}'
+        self.dim_ordering = dim_ordering
+        self.initial_weights = weights
+        super(AbstractConvolution, self).__init__(**kwargs)
+
+    def build(self, input_shape):
+        assert len(input_shape) == 2 + len(self.dimensions)
+        self.input_spec = [InputSpec(shape=input_shape)]
+
+        self.W = self.init(self.W_shape, name='{}_W'.format(self.name))
+        if self.bias:
+            self.b = K.zeros((self.nb_filter,), name='{}_b'.format(self.name))
+            self.trainable_weights = [self.W, self.b]
+        else:
+            self.trainable_weights = [self.W]
+        self.regularizers = []
+
+        if self.W_regularizer:
+            self.W_regularizer.set_param(self.W)
+            self.regularizers.append(self.W_regularizer)
+
+        if self.bias and self.b_regularizer:
+            self.b_regularizer.set_param(self.b)
+            self.regularizers.append(self.b_regularizer)
+
+        if self.activity_regularizer:
+            self.activity_regularizer.set_layer(self)
+            self.regularizers.append(self.activity_regularizer)
+
+        self.constraints = {}
+        if self.W_constraint:
+            self.constraints[self.W] = self.W_constraint
+        if self.bias and self.b_constraint:
+            self.constraints[self.b] = self.b_constraint
+
+        if self.initial_weights is not None:
+            self.set_weights(self.initial_weights)
+            del self.initial_weights
+
+    def get_config(self, **child_class_config):
+        def get_config_if_not_none(x):
+            return x.get_config() if x else None
+
+        return dict(nb_filter=self.nb_filter,
+                    init=self.init.__name__,
+                    activation=self.activation.__name__,
+                    border_mode=self.border_mode,
+                    W_regularizer=get_config_if_not_none(self.W_regularizer),
+                    b_regularizer=get_config_if_not_none(self.b_regularizer),
+                    activity_regularizer=get_config_if_not_none(self.activity_regularizer),
+                    W_constraint=get_config_if_not_none(self.W_constraint),
+                    b_constraint=get_config_if_not_none(self.b_constraint),
+                    bias=self.bias,
+                    **child_class_config)
```

Comment:
btw, the nice things about constructing dicts in this way is that the child class just calls super with the extra bits, and because we are using keyword arguments, python does a check on the arguments that none are duplicated, which catches certain mistakes. Of course, if a child class wanted to override an argument, it can still do so, but it will be explicit, which helps readability. Picked this up from working working with django.